chalice/cli/factory.py

```diff
@@ -0,0 +1,128 @@
+                # When using the ZipFile argument (which is used in chalice),
+                # the entire deployment package zip is sent as a base64 encoded
+                # string.  We don't want this to clutter the debug logs
+                # so we don't log the request body for lambda operations
+                # that have the ZipFile arg.
+                record.args = (record.args[:-1] +
+                               ('(... omitted from logs due to size ...)',))
+        return True
+
+
+class CLIFactory(object):
+    def __init__(self, project_dir, debug=False, profile=None):
+        # type: (str, bool, Optional[str]) -> None
+        self.project_dir = project_dir
+        self.debug = debug
+        self.profile = profile
+
+    def create_botocore_session(self):
+        # type: () -> session.Session
+        return create_botocore_session(profile=self.profile,
+                                       debug=self.debug)
+
+    def create_default_deployer(self, session, prompter):
+        # type: (session.Session, deployer.NoPrompt) -> deployer.Deployer
+        return deployer.create_default_deployer(
+            session=session, prompter=prompter)
+
+    def create_config_obj(self, chalice_stage_name='dev', autogen_policy=True,
+                          api_gateway_stage=None):
+        # type: (str, bool, Optional[str]) -> Config
+        user_provided_params = {}  # type: Dict[str, Any]
+        default_params = {'project_dir': self.project_dir}
+        try:
+            config_from_disk = self.load_project_config()
+        except (OSError, IOError):
+            raise RuntimeError("Unable to load the project config file. "
+                               "Are you sure this is a chalice project?")
+        app_obj = self.load_chalice_app()
+        user_provided_params['chalice_app'] = app_obj
+        if autogen_policy is not None:
+            user_provided_params['autogen_policy'] = autogen_policy
+        if self.profile is not None:
+            user_provided_params['profile'] = self.profile
+        if api_gateway_stage is not None:
+            user_provided_params['api_gateway_stage'] = api_gateway_stage
+        config = Config(chalice_stage_name, user_provided_params,
+                        config_from_disk, default_params)
+        return config
+
+    def create_app_packager(self, config):
+        # type: (Config) -> AppPackager
+        return create_app_packager(config)
+
+    def load_chalice_app(self):
+        # type: () -> Chalice
+        if self.project_dir not in sys.path:
+            sys.path.append(self.project_dir)
+        try:
+            app = importlib.import_module('app')
+            chalice_app = getattr(app, 'app')
+        except Exception as e:
+            # TODO: better error.
+            exception = Exception(
+                "Unable to import your app.py file: %s" % e
+            )
+            # exception.exit_code = 2
```

Comment:
Old code I need to update.  This code was originally from the cli [__init__.py](https://github.com/awslabs/chalice/blob/master/chalice/cli/__init__.py#L101) which used click exceptions to propagate exit codes.